### PR TITLE
Add generic skeleton to preview page

### PR DESCRIPTION
**Pull request by oandregal**
Addresses https://github.com/WordPress/gutenberg/issues/7922

This PR substitutes current text message in the preview page with a generic skeleton as per [Sarah's proposal](https://github.com/WordPress/gutenberg/issues/7922#issuecomment-419123818). If the component is passed any HTML via the `markup` property it'll render that as the skeleton, otherwise it'll render the default skeleton - this opens the opportunity to build a custom skeleton per post.

--- 

_Note that the following GIF doesn't do justice (had to slow down things quite a bit for the framerate to be actually useful), but I guessed it is useful to have a quick glance at how it looks:_

peek 2018-10-15 13-32

### CODE REVIEW

File: packages/editor/src/components/post-preview-button/index.js [hunk 1]
```diff
@@ -89,26 +89,136 @@ export class PostPreviewButton extends Component {
 		this.props.autosave();
 
 		const markup = `
-			<div class="editor-post-preview-button__interstitial-message">
-				<p>Please wait&hellip;</p>
-				<p>Generating preview.</p>
+			<div class="post">
+  				<h1 class="editor-post-title__input"></h1>
```
Comment (tofumatt) [inline packages/editor/src/components/post-preview-button/index.js]: Using multiple empty H1s is semantically not-so-great.

I think it'd make more sense to have this kind of markup:

```html
<h1>Generating Preview</h1>
<img class="cover-image" alt="Placeholder animation for cover image." />
<p>Please wait for preview…</p>
<span class="fake-paragraph"></span>
<span class="fake-header"></span>
<span class="fake-paragraph"></span>
<span class="fake-image"></span>
<span class="fake-list"></span>

[ etc. ]
```

You'll need to create semantically unimportant elements that *look* like paragraphs, lists, etc. or this content will be quite confusing to screenreaders.

**Review by tofumatt — CHANGES_REQUESTED** [review]
There are some changes to the markup and text that should happen here to ensure semantic/accessible content. I also noticed the bug where going "back" in the browser to a preview is stranger now because there is *no* content:



That bug exists in master but it looks like this:



So it would be useful to add *some* text to the skeleton to give context and say what's happening on the page, in case the user goes back to it by mistake. I did that and got confused; it looks like it's loading but never progresses if you go back to it.

File: packages/editor/src/components/post-preview-button/index.js (hunk 1 above)
Comment (tofumatt) [reply packages/editor/src/components/post-preview-button/index.js]: This isn't resolved, but the markup seems to have changed so GitHub thinks it is.

**Comment by tofumatt** [comment]
GitHub thinks: 
https://github.com/WordPress/gutenberg/pull/10613#discussion_r225139359 is outdated, but it's not 😄

**Comment by karmatosed** [comment]
Noting for 5.0 let's focus on something a little simpler, we can always iterate up from this. For discussion see: https://github.com/WordPress/gutenberg/issues/7922#issuecomment-429834509.

**Pull request closed without merging**